validation/tmSchemaGenerator.js

```diff
@@ -79,6 +79,18 @@ function staticReplace(argObject){
     argObject.title = "Thing Model"
     argObject.description = "JSON Schema for validating Thing Models. This is automatically generated from the WoT TD Schema."
     argObject.definitions.type_declaration = {
+        "oneOf": [{
```

Comment:
normal type declaration is not needed anymore right?

Reply:
I actually separated the top level `@type` definition from the one for other levels because otherwise you have to use `tm:ThingModel` in `@type` fields of every level. This new approach, however, removes, allows you to use `tm:ThingModel` in *any* `@type` field. If this should be kept as is we could also just use the existing definition from the TD schema.

Reply:
Oh I see, what you did is indeed correct :)